app/services/mixtape_generator.rb

```diff
@@ -0,0 +1,20 @@
+class MixtapeGenerator
+  def initialize(genre:)
+    @genre = genre
+  end
+
+  def self.perform(genre:)
+    new(genre: genre).perform
+  end
+
+  def perform
+    Mixtape.new(genre: @genre, songs: pick_songs)
```

Comment:
Oh right-- the names of the genre(s) could just be sideloaded from the list of songs. My line of thinking was that the response might contain the requested genre name in addition to the songs themselves, but I see now that this is not necessary as an attribute of ```Mixtape```.